benchmarks/image-classification/rust-benchmark/src/main.rs

```diff
@@ -0,0 +1,103 @@
+use image::io::Reader;
+use image::DynamicImage;
+use sightglass_api as bench;
+use std::convert::TryInto;
+use std::fs;
+use wasi_nn;
+mod imagenet_classes;
+
+pub fn main() {
+    let xml = fs::read_to_string("./mobilenet.xml").unwrap();
+    let weights = fs::read("./mobilenet.bin").unwrap();
+
+    let graph = unsafe {
+        wasi_nn::load(
+            &[&xml.into_bytes(), &weights],
+            wasi_nn::GRAPH_ENCODING_OPENVINO,
+            wasi_nn::EXECUTION_TARGET_CPU,
+        )
+        .unwrap()
+    };
+
+    let context = unsafe { wasi_nn::init_execution_context(graph).unwrap() };
+    let tensor_data = image_to_tensor("test.jpg".to_string(), 224, 224);
+    let tensor = wasi_nn::Tensor {
+        dimensions: &[1, 3, 224, 224],
+        type_: wasi_nn::TENSOR_TYPE_F32,
+        data: &tensor_data,
+    };
+    unsafe {
+        wasi_nn::set_input(context, 0, tensor).unwrap();
+    }
+
+    // Execute the inference.
+    unsafe {
+        bench::start();
+        wasi_nn::compute(context).unwrap();
+        bench::end();
+    }
+
+    // Retrieve the output.
+    let mut output_buffer = vec![0f32; 1001];
+    unsafe {
+        wasi_nn::get_output(
+            context,
+            0,
+            &mut output_buffer[..] as *mut [f32] as *mut u8,
+            (output_buffer.len() * 4).try_into().unwrap(),
+        )
+        .unwrap();
+    }
+
+    let results = sort_results(&output_buffer);
+    eprintln!("{:?}", results[0]);
+
+    for i in 0..5 {
+        eprintln!(
+            "{}.) {}",
+            i + 1,
+            imagenet_classes::IMAGENET_CLASSES[results[i].0]
+        );
+    }
+}
+
+// Sort the buffer of probabilities. The graph places the match probability for each class at the
+// index for that class (e.g. the probability of class 42 is placed at buffer[42]). Here we convert
+// to a wrapping InferenceResult and sort the results.
+fn sort_results(buffer: &[f32]) -> Vec<InferenceResult> {
+    let mut results: Vec<InferenceResult> = buffer
+        .iter()
+        .skip(1)
+        .enumerate()
+        .map(|(c, p)| InferenceResult(c, *p))
+        .collect();
+    results.sort_by(|a, b| b.1.partial_cmp(&a.1).unwrap());
+    results
+}
+
+// Take the image located at 'path', open it, resize it to height x width, and then converts
+// the pixel precision to FP32. The resulting BGR pixel vector is then returned.
+fn image_to_tensor(path: String, height: u32, width: u32) -> Vec<u8> {
+    let pixels = Reader::open(path).unwrap().decode().unwrap();
+    let dyn_img: DynamicImage = pixels.resize_exact(width, height, image::imageops::Triangle);
```

Comment:
Is there an issue anywhere (or TODO) to remind us to replace with the new and improved `image2tensor` crate?

Reply:
I've added a TODO: